### PR TITLE
ramips: Add support for Sercomm S1500 clones

### PR DESCRIPTION
**Pull request by MaxS0niX**
This PR adds support for wireless routers:
OEM/ODM Serсomm S1500 AWI - [Beeline SmartBox PRO](https://openwrt.org/inbox/toh/beeline/smartbox_pro)
OEM/ODM Serсomm S1500 BUC - [WiFire S1500.NBN](https://openwrt.org/toh/wifire/s1500_nbn)

- [Specification](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065#general-specification-2)
- [MAC addresses as verified by OEM firmware](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065/15?u=maxs0nix)

### Installation
* Web UI Update
  1. Rename the firmware so that there are no dots in the name, except
  	for the dot in the ".img" file type. Example "openwrt-factory.img".
  2. Upload and update the firmware via the original web interface.

Remark: You might need make the 2rd step twice if your running firmware
is booted from the Slot 1 (Sercomm0 bootflag). The stock firmware
reverses the bootflag (Sercomm0 / Sercomm1) on each firmware update. 

#### Discussion forum topic:
* [Adding support for Sercomm S1500 clones (Beeline SmartBox Pro, WiFire S1500.NBN)](https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065)
##### Related devices (SerComm OEM/ODM):
* Sercomm S1500 - #4770 
  * [Beeline SmartBox PRO](https://wikidevi.wi-cat.ru/Beeline_SmartBox_Pro)
  * [WiFire S1500.NBN](https://wikidevi.wi-cat.ru/WiFire_S1500.NBN)
* Sercomm S2
  * [Beeline SmartBox GIGA](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-giga-on-stock-u-boot/99390) - :heavy_check_mark: #4195
* Sercomm S3
  * [Beeline SmartBox TURBO+](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-turbo/99635) - #4108
  * [Beeline SmartBox TURBO](https://forum.openwrt.org/t/add-support-for-beeline-smartbox-turbo/99635/37) - :heavy_check_mark: #10118
  * [Etisalat Sercomm S3](https://forum.openwrt.org/t/add-support-for-sercomm-s3-on-stock-uboot/95970)
  * Rostelecom RT-SF-1
  * Rostelecom RT-FE-1

### CODE REVIEW

File: target/linux/ramips/image/mt7621.mk [hunk 1]
```diff
@@ -213,6 +262,38 @@ define Device/asus_rt-n56u-b1
...
+  KERNEL := kernel-bin | append-dtb | lzma | loader-kernel | \
+    lzma | uImage lzma
+  KERNEL_INITRAMFS := kernel-bin | append-dtb | lzma | loader-kernel | \
+    lzma | uImage lzma
+  LOADER_TYPE := bin
+  IMAGES += kernel.bin rootfs.bin factory.img
+  IMAGE/kernel.bin := append-ubi | sercomm-tag-header-kernel 0x1700100 0x1f00000
+  IMAGE/rootfs.bin := append-ubi | check-size
+  IMAGE/factory.img := append-ubi | sercomm-tag-factory-img-pro 0x1700100 0x1f00000 0x1b00100 0x3d00000
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
```
Comment (adschm) [inline target/linux/ramips/image/mt7621.mk:283]: What do we need all these images for? Commit message only mentions sysupgrade and initramfs.

File: target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+&pcie0 {
+	wifi@0,0 { // MT7612EN
+		compatible = "mediatek,mt76";
+		reg = <0x0 0 0 0 0>;
+		ieee80211-freq-limit = <5000000 6000000>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+
+		nvmem-cells = <&macaddr_factory_21000>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <3>;
```
Comment (adschm) [inline target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts:262]: Please add a MAC address overview to the commit message.
https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses

File: target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+	compatible = "beeline,smartbox-pro", "mediatek,mt7621-soc";
+	model = "Beeline SmartBox PRO";
+
+	aliases {
+		label-mac-device = &gmac0;
+
+		led-boot = &status_white;
+		led-failsafe = &wlan2g;
+		led-running = &status_blue;
+		led-upgrade = &wlan5g;
```
Comment (adschm) [inline target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts:18]: If there are status LEDs available, please don't hijack LED with other purposes.

**Comment by adschm** [comment]
Please don't change you new scripts again in each patch. One commit for providing the _proper_ script, one commit for adding the device.

**Comment by MaxS0niX** [comment]
48b2344 - Changes: 
* The repository was updated to the current state of master.
* Correct MAC's.
* Correct bootloader environment.
* ubi-concat

File: target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts (hunk 3 above)
Comment (MaxS0niX) [reply target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts]: Fixed.

File: target/linux/ramips/image/mt7621.mk (hunk 1 above)
Comment (MaxS0niX) [reply target/linux/ramips/image/mt7621.mk:283]: Left for the duration of the tests. Then leave only factory & sysupgrade?

File: target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts (hunk 2 above)
Comment (MaxS0niX) [reply target/linux/ramips/dts/mt7621_beeline_smartbox-pro.dts]: Added to commit. Also described here.
https://forum.openwrt.org/t/adding-support-for-sercomm-s1500-clones-beeline-smartbox-pro-wifire-s1500-nbn/110065/15?u=maxs0nix

**Comment by MaxS0niX** [comment]
cad82b5 - Changes: 
* The repository was updated to the current state of master.
* LED functions moved to 01_leds
* "ROUT<->REP Switch_bt" turns off Wi-Fi
* Used firmware-utils Sercomm

There are several questions:
1) Is it possible to issue a Pull Request to the firmware-utils repository via github?
2) LEDs do not work for Link 1Gb/s event. How to fix?)
3) Is it possible to add to the device in the next v22.03.x release?
4) There is a similar device [WiFire S1500.NBN](https://gitlab.com/openwrt-dev-ru-smartbox/openwrt-mt7621-sercomm-smartbox/-/tree/sercomm_s1500) is it better to add it in this PR with a common dtsi or a separate one?

**Comment by hauke** [comment]
> Is it possible to issue a Pull Request to the firmware-utils repository via github?
No, you have to send a patch to the openwrt devel malling list
> Is it possible to add to the device in the next v22.03.x release?
It should first go into master
> There is a similar device [WiFire S1500.NBN](https://gitlab.com/openwrt-dev-ru-smartbox/openwrt-mt7621-sercomm-smartbox/-/tree/sercomm_s1500) is it better to add it in this PR with a common dtsi or a separate one?
If you have a common dtsi file it is probably easier to add it in one pull request, but in different commits.

**Comment by hauke** [comment]
This should be fine:
1) Common dtsi
2) dts PRO + changes in mt7621.mk 
3) dts NBN + changes in mt7621.mk

**Comment by MaxS0niX** [comment]
> This should be fine:
> 
>     1. Common dtsi
> 
>     2. dts PRO + changes in mt7621.mk
> 
>     3. dts NBN + changes in mt7621.mk

Thanks for the clarification!

**Comment by neheb** [comment]
Only thing I see is the lack of GPIO controlled switch LEDs.

Please adjust to be compatible with https://github.com/openwrt/openwrt/pull/10238

**Pull request closed without merging**